test/inductor/test_torchinductor.py

```diff
@@ -2890,6 +2890,19 @@ def forward(x, y):
         y = torch.tensor([[[0.6331]], [[1.6358]], [[-0.3459]], [[1.0196]]])
         self.common(forward, (x, y))
 
+    def test_div_softmax_symint(self):
+        def forward(x, y):
+            z = x.div(y * x.shape[-1])
+            return F.softmax(z, dim=-1)
+
+        query = torch.randn(1, 10, 40)
+        key = torch.randn(1, 2, 40)
+        x = torch.matmul(query, key.transpose(-2, -1))
+
+        cf = torch.compile(forward, dynamic=True)
```

Comment:
How did this fail previously ?

Reply:
Failed with a
```
torch._dynamo.exc.BackendCompilerFailed: backend='inductor' raised:
TypeError: where() received an invalid combination of arguments - got (SymBool, FakeTensor, FakeTensor), but expected one of:
 * (Tensor condition)
 * (Tensor condition, Tensor input, Tensor other, *, Tensor out = None)
 * (Tensor condition, Number self, Tensor other)
      didn't match because some of the arguments have invalid types: (SymBool, FakeTensor, FakeTensor)
 * (Tensor condition, Tensor input, Number other)
      didn't match because some of the arguments have invalid types: (SymBool, FakeTensor, FakeTensor)
 * (Tensor condition, Number self, Number other)
      didn't match because some of the arguments have invalid types: (SymBool, FakeTensor, FakeTensor)
```